src/parser/column_definition.cpp

```diff
@@ -1,39 +1,206 @@
+const storage_t &ColumnDefinition::StorageOid() const {
+	return storage_oid;
+}
+
+void ColumnDefinition::SetStorageOid(storage_t storage_oid) {
+	this->storage_oid = storage_oid;
+}
+
+const column_t &ColumnDefinition::Oid() const {
+	return oid;
+}
+
+void ColumnDefinition::SetOid(column_t oid) {
+	this->oid = oid;
+}
+
+const TableColumnType &ColumnDefinition::Category() const {
+	return category;
+}
+
+bool ColumnDefinition::Generated() const {
+	return category == TableColumnType::GENERATED;
+}
+
+//===--------------------------------------------------------------------===//
+// Generated Columns (VIRTUAL)
+//===--------------------------------------------------------------------===//
+
+static void VerifyColumnRefs(ParsedExpression &expr) {
+	if (expr.type == ExpressionType::COLUMN_REF) {
+		auto &column_ref = (ColumnRefExpression &)expr;
+		if (column_ref.IsQualified()) {
+			throw ParserException(
+			    "Qualified (tbl.name) column references are not allowed inside of generated column expressions");
+		}
+	}
+	ParsedExpressionIterator::EnumerateChildren(
+	    expr, [&](const ParsedExpression &child) { VerifyColumnRefs((ParsedExpression &)child); });
+}
+
+static void InnerGetListOfDependencies(ParsedExpression &expr, vector<string> &dependencies) {
+	if (expr.type == ExpressionType::COLUMN_REF) {
+		auto columnref = (ColumnRefExpression &)expr;
+		auto &name = columnref.GetColumnName();
+		dependencies.push_back(name);
+	}
+	ParsedExpressionIterator::EnumerateChildren(expr, [&](const ParsedExpression &child) {
+		InnerGetListOfDependencies((ParsedExpression &)child, dependencies);
+	});
+}
+
+void ColumnDefinition::GetListOfDependencies(vector<string> &dependencies) const {
+	D_ASSERT(Generated());
+	InnerGetListOfDependencies(*generated_expression, dependencies);
+}
+
+void ColumnDefinition::SetGeneratedExpression(unique_ptr<ParsedExpression> expression) {
+	category = TableColumnType::GENERATED;
+
+	if (expression->HasSubquery()) {
+		throw ParserException("Expression of generated column \"%s\" contains a subquery, which isn't allowed", name);
+	}
+
+	VerifyColumnRefs(*expression);
+	if (type.id() == LogicalTypeId::ANY) {
+		generated_expression = move(expression);
+		return;
+	}
+	// Always wrap the expression in a cast, that way we can always update the cast when we change the type
+	// Except if the type is LogicalType::ANY (no type specified)
+	generated_expression = make_unique_base<ParsedExpression, CastExpression>(type, move(expression));
+}
+
+void ColumnDefinition::ChangeGeneratedExpressionType(const LogicalType &type) {
+	D_ASSERT(Generated());
+	// First time the type is set, add a cast around the expression
+	if (this->type.id() == LogicalTypeId::ANY) {
+		generated_expression = make_unique_base<ParsedExpression, CastExpression>(type, move(generated_expression));
+		return;
+	}
+	// Every generated expression should be wrapped in a cast on creation
+	D_ASSERT(generated_expression->type == ExpressionType::OPERATOR_CAST);
```

Comment:
This code is never run

Reply:
This was left over from implementing type-change for generated columns, which we've disabled for now, but this code could proof useful in the future, so I left it, is that alright?